src/components/MyList/MyList.tsx

```diff
@@ -0,0 +1,38 @@
+import { memo, useMemo } from 'react';
+import { User } from 'models/user';
+import { MyListItem } from 'components/MyListItem/MyListItem';
+import './MyList.css';
+
+export interface MyListProps {
+  data?: User[];
+  loading: boolean;
+  error?: string;
+}
+
+function MyList(props: MyListProps): JSX.Element {
+  const { data, loading, error } = props;
+  const listItems = useMemo(() => {
+    return data?.map((user) => {
+      return <MyListItem key={user.name} user={user} />;
+    });
+  }, [data]);
+
+  if (loading) return <> Loading </>;
+  if (error) return <> Error: {error} </>;
```

Comment:
Usually, this loading should be handled in the parent, so your component can be simpler